DungeonFinder/users/views.py

```diff
@@ -119,3 +119,6 @@ class GMSignUp:
 
 class GMProfileUpdate:
     pass
+
+def themes(request):
```

Comment:
Okay so it has to be a class based view, as we're processing both GET and POST requests. And you might want to use a form for validation.

First, write the view:

```python
"""
We inherit from two classes here.
- LoginRequiredMixin means that someone has to be logged in to view.
- DFEditView here is what we inherit from if we are editing a model
"""
class UserUpdateTheme(LoginRequiredMixin, DFEditView):
    model = User   # The model that will be edited
    form_class = UserUpdateThemeForm  # The form that we'll insert into the template and validate with
    template_name = 'foobar.jinja'  # Your bit

    def get_object(self, queryset=None):
        """
        Normally, this would be used to return the object (like the Campaign) we're editing. In this case though,
        we want to make sure the object being edited is the logged in user.
        """
        return self.request.user  

    def form_valid(self, form):
        """
        This is called when you POST, and the data is validated against the form.
        """
        theme = form.cleaned_data['theme']
        self.request.user.theme = theme  # Get the data from the form
        self.request.user.save()  # and save
        messages.success(request, 'Well done, you did good editing')  # Displaying the nice message to the user
        return redirect('dashboard')  # Not sure where you want to return the user to after, but this is where you say


user_update_theme = UserUpdateTheme.as_view()
```
Now the form, this should go in `users/forms.py`

```python
class UserUpdateThemeForm(DFForm):  # As with the view, we inherit from our base form class.
    # choices here should be a list of tuples. So like [('choice_1_value', 'Choice 1 label'), ('choice_2_value', 'Choice 2 label')]
    theme = forms.ChoiceField(choices=your_choices)
```

Lastly, in the template:
```jinja
  <form method="post">
    <input type="hidden" name="csrfmiddlewaretoken" value="{{ csrf_token }}">
    <div id="theme_form">
      {{ form }}  # This will give you your select input with id `id_theme`
      <button class="btn btn-primary" type="submit">Save</button>
    </div>
  </form>
```